src/Providers/ElasticApmServiceProvider.php

```
@@ -225,12 +225,15 @@ public static function getGuzzleMiddleware() : callable
 				function(RequestInterface $request, array $options) {
 					self::$lastHttpRequestStart = microtime(true);
 				},
-				function (RequestInterface $request, array $options, FulfilledPromise $response) {
+				function (RequestInterface $request, array $options, PromiseInterface $promise) {
 					// leave early if monitoring is disabled
 					if (config('elastic-apm.active') !== true || config('elastic-apm.spans.httplog.enabled') !== true) {
 						return;
 					}
 
+					/* @var $response \GuzzleHttp\Psr7\Response */
+					$response = $promise->wait(true);
```

>Pass $unwrap as true to unwrap the result of the promise, either returning the resolved value or throwing the rejected exception.

Are you sure we want to throw the exception here?

---

Failed IWAN request throws an exception later on, but good point - let's silently skip it here.